consensus/proto_array/src/proto_array.rs

```diff
@@ -126,10 +129,17 @@ impl ProtoArray {
                 continue;
             }
 
-            let mut node_delta = deltas
-                .get(node_index)
-                .copied()
-                .ok_or(Error::InvalidNodeDelta(node_index))?;
+            let mut node_delta = if node.execution_status.is_invalid() {
+                // If the node has an invalid execution payload, reduce its weight to zero.
+                0_i64
+                    .checked_sub(node.weight as i64)
+                    .ok_or(Error::InvalidExecutionDeltaOverflow(node_index))?
```

Comment:
Ah yes, great point. I've addressed this over in 4e3a078

Perhaps @michaelsproul could give that commit a once-over too, since he's worked with proposer boost? :pray:

Reply:
LGTM!